packages/compiler-core/src/parse.ts

```diff
@@ -299,6 +299,16 @@ function parseChildren(
           node.content = node.content.replace(/\r\n/g, '\n')
         }
       }
+      // #9229 innerHTML normalize \r\n or \r into a single \n in the DOM
+      else if (
+        node.type === NodeTypes.INTERPOLATION &&
+        node.content.type === NodeTypes.SIMPLE_EXPRESSION
+      ) {
+        node.content.content = node.content.content.replace(
+          /\\r\\n|\\r/g,
+          '\\n'
+        )
+      }
```

Comment:
I think this only fixes the problem for string literals included directly in the template. I'd imagine the more common scenario in real code is for the `\r` to be in the data:

- [Playground](https://deploy-preview-9232--vue-sfc-playground.netlify.app/#__DEV____SSR__eNp9kj1PwzAQhv/KyUuX0qqCKUorAeoAAyBg4xii9NqkOGfLdtpKUf47FweaFqFu8ftxec5yo26tnexqUolKfe5KG8BTqO0COTfsA+SZJw9z+EAGGLkEoCCtDaLbG6dXo3Gvc3LUEfnMks5gnbV4sPivNfzoZxryJ3I67SEFTw6BKquzQHICSIsZ7K7Wxs1R5VByT45q0TSQQ9um02ImwXR60lJjFbzsuS43k603LLfQdLNkgqlsqck921DKPaBKIDqdlwnW/jFqwdUUgWOnoPzrH33rD52G6sWRJ7cjVEcvZG5DobeXb090kO+jWZlVrSV9wXwlb3TdMfaxu5pXgn2Si7QPlTUulLx598tDIPa/S3WgXbKNeVTyFO4vrD7gXk9uYg+5Ve0362TA/w==)

Reply:
Done~